docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+- **Custom Metadata Tags:** Utilize custom metadata to mark specific objects (snapshots) that should be bypassed. To exclude a snapshot from the restoration process, attach custom metadata to it with the key `x-etcd-snapshot-exclude` and value `true`. This method is officially supported, as demonstrated in the [etcd-backup-restore PR](https://github.com/gardener/etcd-backup-restore/pull/776).
+
+## Compatibility
+
+The proposed changes are fully compatible with existing ETCD clusters and backup processes. Operators are responsible for creating or updating the immutability settings on the backup storage buckets, but no changes are required to the ETCD clusters themselves.
+
+- **Backward Compatibility:** Existing clusters without immutable buckets will continue to function without change.
+- **Forward Compatibility:** Clusters can opt-in to use immutable backups by configuring the bucket accordingly.
+
+## Implementation Steps
+
+1. **Enhance the Compaction Job:**
+     - Modify the compaction job in `etcd-backup-restore` to manage backup processes for hibernated clusters, including snapshot creation and garbage collection, while considering the new immutability constraints.
+     - Implement the `--hibernation-snapshot-interval` flag in the compaction controller. Ensure that the compaction job can start an embedded ETCD instance to take snapshots during hibernation.
+
+2. **Update Documentation:**
+     - Revise the documentation to reflect the changes and guide operators in effectively using the new immutability features.
+     - Provide detailed guidelines on configuring buckets with immutability settings.
+     - Document procedures for excluding snapshots when necessary.
+
+## Risks and Mitigations
+
+- **Increased Storage Costs:**
+
+  - **Risk:** Introducing immutability could lead to increased storage costs due to the inability to delete backups before the immutability period ends.
+  - **Mitigation:** Operators should carefully configure immutability periods and monitor storage utilization. Garbage collection will help mitigate long-term storage growth.
+
+- **Backup Gaps During Hibernation:**
+
+  - **Risk:** Hibernated clusters might not receive necessary backups, potentially leading to compliance issues.
+  - **Mitigation:** The compaction job's enhancement ensures that backups are taken during hibernation at configured intervals.
+
+- **Excluding Critical Snapshots:**
+
+  - **Risk:** Excluding snapshots during restoration might be misused or lead to incomplete data restoration.
+  - **Mitigation:** Restrict the ability to tag snapshots for exclusion to authorized personnel. Implement audit logging for actions that tag snapshots.
+
+## Operational Considerations
+
+Operators need to:
+
+- **Bucket Configuration:**
+
+  - Configure buckets with appropriate immutability settings before deploying ETCD clusters.
+  - Ensure that the immutability periods align with organizational policies.
+
+- **Compaction Job Configuration:**
```

Comment:
What is the retry threshold for this job? 
What happens if it fails to run for a period of 24hrs. 
What happens if druid is down?
What happens when druid comes back up esp. for failed jobs which have breached the retry threshold? 
What happens if we breach the bucket retention period? Is no data to restore possible on wake-up of hibernated clusters. 
Does garbage collection runs independent or in sequence only after the job takes a full snapshot on its run.